### PR TITLE
Add new table for features extensions

### PR DESCRIPTION
**Pull request by xxbillouxx**

### CODE REVIEW

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

**Comment by xxbillouxx** [comment]
recheck

File: alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    op.create_table(
+        'feature_extension',
+        sa.Column('uuid', UUID, server_default=sa.text('uuid_generate_v4()'),
+                  primary_key=True),
+        sa.Column('enabled', sa.Boolean, nullable=False, server_default='true'),
+        sa.Column('exten', sa.String(40), nullable=False),
+        sa.Column('feature', sa.String(255), nullable=False),
+    )
+
+    op.create_index(
```
Comment (pc-m) [inline alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py:29]: The index on the primary key is redundant. Postgresql creates an index for all primary keys

File: alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def downgrade():
+    op.execute('''
+    INSERT INTO extensions (commented, context, exten, type, typeval)
+      SELECT 
+        CASE enabled WHEN False THEN 1 ELSE 0 END as commented,
+        TEXT 'xivo-features' as context,
+        exten,
+        TEXT 'extenfeatures' as type
```
Comment (pc-m) [inline alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py:79]: Syntax error here
I think the line `TEXT 'extenfeatures' as type` needs a coma and the line `feature as typeval,` does not

Comment (pc-m) [reply alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py:79]: There's also a type error
```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DatatypeMismatch) column "type" is of type extenumbers_type but expression is of type text
LINE 7:         TEXT 'extenfeatures' as type,
```

**Review by pc-m — CHANGES_REQUESTED** [review]

File: alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py (hunk 1 above)
Comment (xxbillouxx) [reply alembic/versions/1d5d4e41d708_move_features_extensions_dedicated_table.py]: Explicit index removed

**Comment by pc-m** [comment]
Looks good to me

**Review by pc-m — DISMISSED** [review]

**Review by fblackburn1 — CHANGES_REQUESTED** [review]
don't forget to rebase alembic version (we should add a zuul check on this)

**Comment by pc-m** [comment]
> don't forget to rebase alembic version (we should add a zuul check on this)

Wouldn't the `debian-packaging-bullseye` check fail during the gate?

**Review by pc-m — APPROVED** [review]

**Comment by fblackburn1** [comment]
> > don't forget to rebase alembic version (we should add a zuul check on this)
> 
> Wouldn't the `debian-packaging-bullseye` check fail during the gate?

Oh maybe :man_shrugging:

**Pull request merged**